splice_si_api/src/main/java/com/splicemachine/si/impl/store/IgnoreTxnSupplier.java

```diff
@@ -0,0 +1,114 @@
+import com.splicemachine.storage.*;
+import com.splicemachine.utils.SpliceLogUtils;
+import org.apache.hadoop.hbase.util.Pair;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by jyuan on 10/30/17.
+ */
+public class IgnoreTxnSupplier {
+    private Set<Pair<Long, Long>> cache;
+    private EntryDecoder entryDecoder;
+    final private PartitionFactory partitionFactory;
+    final private TxnOperationFactory txnOperationFactory;
+    private boolean ignoreTxnTableExists;
+    private boolean initialized = false;
+
+    public IgnoreTxnSupplier(PartitionFactory partitionFactory, TxnOperationFactory txnOperationFactory) {
+        this.partitionFactory = partitionFactory;
+        this.txnOperationFactory = txnOperationFactory;
+        cache = new HashSet<>();
+    }
+
+    public boolean shouldIgnore(Long txnId) throws IOException{
+
+        boolean ignore = false;
+
+        init();
+        
+        if (!cache.isEmpty()) {
+            for (Pair<Long, Long> range : cache) {
+                if (txnId > range.getFirst() && txnId < range.getSecond()) {
+                    ignore = true;
+                    break;
+                }
+            }
+        }
+
+        return ignore;
+    }
+
+
+    private void populateIgnoreTxnCache() throws IOException {
+
+        PartitionAdmin admin = partitionFactory.getAdmin();
+        ignoreTxnTableExists = admin.tableExists(HBaseConfiguration.IGNORE_TXN_TABLE_NAME);
+        if (ignoreTxnTableExists) {
+            if (entryDecoder == null)
+                entryDecoder = new EntryDecoder();
+            DataResultScanner scanner = null;
+            try {
+                scanner = openScanner();
+                DataResult r = null;
+                while ((r = scanner.next()) != null) {
+                    DataCell cell = r.userData();
+                    byte[] buffer = cell.valueArray();
+                    int offset = cell.valueOffset();
+                    int length = cell.valueLength();
+                    entryDecoder.set(buffer, offset, length);
+                    MultiFieldDecoder decoder = entryDecoder.getEntryDecoder();
+                    long startTxnId = decoder.decodeNextLong();
+                    long endTxnId = decoder.decodeNextLong();
+                    cache.add(new Pair<Long, Long>(startTxnId, endTxnId));
+                }
+            } finally {
+                if (scanner != null)
+                    scanner.close();
+            }
+        }
+    }
+
+    private DataResultScanner openScanner() throws IOException {
+        Partition table = partitionFactory.getTable(HBaseConfiguration.IGNORE_TXN_TABLE_NAME);
+        DataScan scan = txnOperationFactory.newDataScan(null);
+        return table.openResultScanner(scan);
+    }
+
+    private void init() throws IOException {
+        if (!initialized) {
```

Comment:
'initialized' needs to be declared 'volatile' to be used in double-checked locking.
Is lazy initialization justified here?